examples/apps/generic_client/generic_client.py

```diff
@@ -19,465 +19,267 @@
-    # This function will send a WorkOrderReceiptCreate JSON RPC request.
-    wo_request = json.loads(wo_params.to_jrpc_string(jrpc_req_id))
-    wo_receipt_request_obj = WorkOrderReceiptRequest()
-    wo_create_receipt = wo_receipt_request_obj.create_receipt(
-        wo_request,
-        ReceiptCreateStatus.PENDING.value,
-        client_private_key
-    )
-    logger.info("Work order create receipt request : {} \n \n ".format(
-        json.dumps(wo_create_receipt, indent=4)
-    ))
-    # Submit work order create receipt jrpc request
-    wo_receipt_resp = wo_receipt.work_order_receipt_create(
-        wo_create_receipt["workOrderId"],
-        wo_create_receipt["workerServiceId"],
-        wo_create_receipt["workerId"],
-        wo_create_receipt["requesterId"],
-        wo_create_receipt["receiptCreateStatus"],
-        wo_create_receipt["workOrderRequestHash"],
-        wo_create_receipt["requesterGeneratedNonce"],
-        wo_create_receipt["requesterSignature"],
-        wo_create_receipt["signatureRules"],
-        wo_create_receipt["receiptVerificationKey"],
-        jrpc_req_id
-    )
-    logger.info("Work order create receipt response : {} \n \n ".format(
-        wo_receipt_resp
-    ))
+class GenericClient():
+    """
+    Generic client class to test end to end flow
+    for both direct model and proxy model
+    """
 
+    def __init__(self, args):
+        # Parse command line arguments
+        options = self._parse_command_line(args)
+        # Read config params
+        if options.config:
+            self._config = self._parse_config_file(options.config)
+        else:
+            self._config = self._parse_config_file(
+                TCF_HOME +
+                "/sdk/avalon_sdk/tcf_connector.toml")
+        if self._config is None:
+            logging.error("\n Error in parsing config file: {}\n")
+            sys.exit(-1)
+        if options.blockchain:
+            self._config['blockchain']['type'] = options.blockchain
+            self._blockchain = options.blockchain
+        else:
+            self._blockchain = self._config['blockchain']['type']
+
+        self._is_direct_mode = False
+        # Mode
+        self._mode = options.mode
+
+        # Http JSON RPC listener uri
+        self._uri = options.uri
+        if self._uri:
+            self._config["tcf"]["json_rpc_uri"] = self._uri
+
+        # Address of smart contract
+        self._address = options.address
+        if self._address:
+            if self._mode == "listing":
+                self._config["ethereum"]["direct_registry_contract_address"] =\
+                    self._address
+            elif self._mode == "registry":
+                logging.error(
+                    "\n Only Worker registry listing address is supported." +
+                    "Worker registry address is unsupported \n")
+                sys.exit(-1)
+
+        bc_type = self._blockchain.lower()
```

Comment:
Please verify if at all this is required as -b is having only [2 specific choices](https://github.com/hyperledger/avalon/pull/618/files?file-filters%5B%5D=.py&hide-deleted-files=true#diff-276a76ec14608a7865df8aab6eedc707R143). So we should strictly expect only 'fabric' or 'ethereum' here, that is in lower case.

Reply:
That is taken care while argparser, it will throw error if we pass wrong value.